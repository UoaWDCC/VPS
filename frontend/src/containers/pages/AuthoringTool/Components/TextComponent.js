import { Typography } from "@material-ui/core";
import React, { useContext } from "react";
import AuthoringToolContext from "context/AuthoringToolContext";
import useStyles from "./components.styles";
// import TextPropertiesComponent from "../CanvasSideBar/ComponentProperties/TextPropertiesComponent";

/**
 * This component represents a text scene component
 * @component
 * @example
 * <TextComponent
 *    id={index}
 *    onClick={onClick}
 *    component={component}
 *    onDoubleClick={handleDoubleClick}
 * />
 */
export default function TextComponent({ id, onClick, component }) {
  const styles = useStyles(component);

  const context = useContext(AuthoringToolContext);
  const textRef = context?.propertiesRefs?.text;

  return (
    <Typography
      className={`${styles.defaultComponentStyling} ${styles.textComponentStyles}`}
      id={id}
      onClick={onClick}
      onDoubleClick={() => textRef?.current?.focus()}
    >
      {component.text}
    </Typography>
  );
}
