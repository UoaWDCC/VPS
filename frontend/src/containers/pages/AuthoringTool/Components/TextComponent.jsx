import { Typography } from "@material-ui/core";
import AuthoringToolContext from "context/AuthoringToolContext";
import { useContext } from "react";
import useStyles from "./components.styles";
// import TextPropertiesComponent from "../CanvasSideBar/ComponentProperties/TextPropertiesComponent";

/**
 * This component represents a text scene component
 * @component
 * @example
 * <TextComponent
 *    id={index}
 *    onClick={onClick}
 *    component={component}
 *    onDoubleClick={handleDoubleClick}
 * />
 */
export default function TextComponent({ id, onClick, component }) {
  const styles = useStyles(component);

  const context = useContext(AuthoringToolContext);
  const textRef = context?.propertiesRefs?.text;

  return (
    <Typography
      className={`${styles.defaultComponentStyling} ${styles.textComponentStyles}`}
      id={id}
      onClick={onClick}
      onDoubleClick={() => textRef?.current?.focus()}
      style={{ zIndex: component?.zPosition || "0" }}
    >
      {component.text}
    </Typography>
  );
}
