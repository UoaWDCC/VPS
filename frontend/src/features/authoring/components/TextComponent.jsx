import { Typography } from "@material-ui/core";
import AuthoringToolContext from "context/AuthoringToolContext";
import { useContext } from "react";
import useStyles from "./components.styles";

/**
 * This component represents a text scene component
 * @component
 * @example
 * <TextComponent
 *    id={index}
 *    onClick={onClick}
 *    component={component}
 *    onDoubleClick={handleDoubleClick}
 * />
 */
export default function TextComponent({ id, onClick, component, zoomLevel }) {
  const styles = useStyles(component);
  const userDefinedWidth = component.width;
  const userDefinedHeight = component.height;

  const context = useContext(AuthoringToolContext);
  const textRef = context?.propertiesRefs?.text;

  return (
    <Typography
      className={`${styles.defaultComponentStyling} ${styles.textComponentStyles}`}
      id={id}
      onClick={onClick}
      onDoubleClick={() => textRef?.current?.focus()}
      style={{
        zIndex: component?.zPosition || "0",
        width: `${userDefinedWidth * zoomLevel}%`,
        height: `${userDefinedHeight * zoomLevel}%`,
      }}
    >
      {component.text}
    </Typography>
  );
}
