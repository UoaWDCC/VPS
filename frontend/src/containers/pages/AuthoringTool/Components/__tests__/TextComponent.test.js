import React from "react";
import { render } from "@testing-library/react";
import TextComponent from "../TextComponent";

test("Add text component snapshot test", () => {
  const { baseElement } = render(
    <TextComponent
      component={{
        type: "TEXT",
        text: "default text",
        left: 50,
        top: 50,
        height: 10,
        width: 20,
      }}
    />
  );

  expect(baseElement).toMatchSnapshot();
});
