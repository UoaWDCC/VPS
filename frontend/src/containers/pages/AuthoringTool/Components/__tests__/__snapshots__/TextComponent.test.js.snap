// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Add text component snapshot test 1`] = `
<body>
  <div>
    <p
      class="MuiTypography-root makeStyles-defaultComponentStyling-1 makeStyles-textComponentStyles-2 MuiTypography-body1"
      style="z-index: 0;"
    >
      default text
    </p>
  </div>
</body>
`;
